source/cli.js

```diff
@@ -91,11 +91,25 @@ updateNotifier({pkg: cli.pkg}).notify();
 		...cli.flags
 	};
 
-	const isAvailable = flags.publish ? await isPackageNameAvailable(pkg) : false;
+	const status = {
```

Comment:
Looks good, though I'd still move this to a separate utility method in `source/npm/util.js` and use that here.

Reply:
I moved it into the existing method `isPackageNameAvailable`; if you'd prefer a separate method entirely, happy to make the change.

Reply:
Yeah, that's be preferable.

Reply:
In hindsight @itaisteinherz , could I trouble you for brief example of what you have in mind, re: separate method? Currently the status of `unknown` is flagged only when the existing method returns an uncaught error, hence the coupling.

Reply:
I'm really sorry, I was looking at an older version of the code, and missed all of the new improvements you made. Thanks for bearing with me on this, and sorry for troubling you 🙏🏻